receiver/mongodbatlasreceiver/receiver.go

```diff
@@ -231,3 +240,19 @@ func (s *receiver) extractProcessDiskMetrics(
 	}
 	return nil
 }
+
+// Log receiver logic
+func (s *receiver) Start(ctx context.Context, host component.Host) error {
```

Comment:
This function, and many others you've added, act as if this `receiver` struct is only concerned with logs. We need to maintain clearer separation of concerns. I suggest creating a separate struct called `logsReceiver`, similar to the `alertsReceiver`.

Reply:
You may need to combine them all together under one struct, but having each as a separate struct to begin with will keep things organized.

The immediate problem I see here is that `Start` and `Stop` are called regardless of whether this is a logs or metrics receiver. So you may have some broken functionality running if they only configured a metrics receiver.